core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -698,6 +663,42 @@ object StorageTool extends Logging {
     voterSet
   }
 
+  def createStandaloneVoterMap(config: KafkaConfig): util.Map[ListenerName, InetSocketAddress] = {
+    val advertisedListenerEndpoints = config.effectiveAdvertisedControllerListeners
+    val listeners: util.Map[ListenerName, InetSocketAddress] = new util.HashMap()
+    advertisedListenerEndpoints.foreach(endpoint => {
+      val host: String = endpoint.host
+      listeners.put(endpoint.listenerName, new InetSocketAddress(host, endpoint.port))
+    })
+    listeners
+  }
+
+  private def parseControllerQuorumVotersMap(controllerQuorumVoterMap: util.Map[Integer, InetSocketAddress],
+                           metaProperties: MetaProperties,
+                           config: KafkaConfig): util.Map[ListenerName, InetSocketAddress] = {
+    val listeners: util.Map[ListenerName, InetSocketAddress] = new util.HashMap()
+    controllerQuorumVoterMap.keySet().forEach(replicaId => {
+      if (metaProperties.nodeId().getAsInt == replicaId) {
+        val listenerNameOption = config.effectiveAdvertisedControllerListeners.
```

Comment:
We should assume that the listener name is the default listener name. The default listener name is the first listener in https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaConfig.scala#L730-L737.

There is also this code for an example of getting the first listener name: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/raft/RaftManager.scala#L267

There is one validation that we should do for the local replica. The local replica's default listener (name, host and port) matches the entry specified in `--controller-quorum-voters`.